packages/medusa/src/loaders/subscribers.ts

```diff
@@ -2,21 +2,24 @@ import glob from "glob"
 import path from "path"
 import { asFunction } from "awilix"
 import { MedusaContainer } from "../types/global"
+import { MedusaError } from "@medusajs/utils"
 
 /**
  * Registers all subscribers in the subscribers directory
  */
 export default ({ container }: { container: MedusaContainer }) => {
-  const isTest = process.env.NODE_ENV === "test"
-
-  const corePath = isTest
-    ? "../subscribers/__mocks__/*.js"
-    : "../subscribers/*.js"
+  const corePath = "../subscribers/*.js"
```

Comment:
Having different implementations of the subscribers beats the purpose of testing them, so I'd rather have us find a good way to wait for the subscribers' completion.